sources/packages/web/src/views/institution/InstitutionUserProfile.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <Message severity="info">
+    Please notice that the read-only information below is retrieved from your
+    BCeID account and it is not possible to change it here. If any read-only
+    information needs to be changed please visit
+    <a rel="noopener" href="https://www.bceid.ca/" target="_blank">bceid.ca</a>.
+  </Message>
+  <Card class="p-m-4">
+    <template #content>
+      <formio
+        formName="institutionUserProfile"
+        :data="initialData"
+        @submitted="submitted"
+      ></formio>
+    </template>
+  </Card>
+</template>
+
+<script lang="ts">
+import { ref, onMounted } from "vue";
+import { useRouter } from "vue-router";
+import { useToast } from "primevue/usetoast";
+import formio from "../../components/generic/formio.vue";
+import { UserService } from "../../services/UserService";
+import { InstitutionUserDetailsDto } from "../../types";
+import { InstitutionRoutesConst } from "../../constants/routes/RouteConstants";
+
+export default {
+  components: { formio },
+  setup() {
+    // Hooks
+    const toast = useToast();
+    const router = useRouter();
+    // Data-bind
+    const initialData = ref({});
+
+    const submitted = async (data: InstitutionUserDetailsDto) => {
+      let redirectHome = true;
+      try {
+        await UserService.shared.updateInstitutionUser(data);
+        toast.add({
+          severity: "success",
+          summary: "Updated!",
+          detail: "Institution User successfully updated!",
+          life: 5000,
+        });
+      } catch (excp) {
+        redirectHome = false;
+        toast.add({
+          severity: "error",
+          summary: "Unexpected error",
+          detail: "An error happened during the update process.",
+          life: 5000,
+        });
+      }
+
+      if (redirectHome) {
```

Comment:
IMO, I think you can avoid this check and add the redirect logic inside the try

Reply:
As discussed the redirect logic is removed from condition.